articles/how-to-build-a-speedtyping-game-using-javascript/index.md

```diff
@@ -0,0 +1,327 @@
+so we should have this;
+
+```Javascript
+quoteInputElement.addEventListener('input', () => {
+  const arrayQuote = quoteDisplayElement.querySelectorAll('span')
+  const arrayValue = quoteInputElement.value.split('')
+ 
+  let correct = true
+  arrayQuote.forEach((characterSpan, index) => {
+    const character = arrayValue[index]
+    if (character == null) {
+        characterSpan.classList.remove('correct')
+      characterSpan.classList.remove('incorrect')
+      correct = false
+    } else if (character === characterSpan.innerText) {
+      characterSpan.classList.add('correct')
+      characterSpan.classList.remove('incorrect')
+    } else {
+      characterSpan.classList.remove('correct')
+      characterSpan.classList.add('incorrect')
+      correct = false
+    }
+  })
+```
+
+All we have left to do is to fix our `timer`.  it’s actually really simple. We would create a `startTimer` `function` and get our `timer` element. If you remember, we created an element with the `id` of `timer` in our HTML, so we would just select the `id` of `timer`.
+
+```Javascript
+const timerElement = document.getElementById('timer')
+```
+
+Now, we would set the `timer` to 'zero' ;
+
+```Javascript
+function startTimer() {
+  timerElement.innerText = 0
+```
+
+Then we call the `startTimer` in our `renderNewQuote`. Whenever a new quote is rendered we want our `timer` to restart. So inside this timer what we would do is,   for every second that we want our timer to be updated, `setInterval` will be used
+
+```Javascript
+setInterval(() => {
+    timer.innerText = getTimerTime()
+  }, 1000)
+```
+
+since it accepts a `function` and a second parameter specifying the number of times you want the `function` to run in milliseconds. So the `function` is going to run every second, but it has a slight problem . `setInterval` isn't precise; it won't run exactly once per second. To compensate for the fact that `setInterval` is inaccurate, we will create a variable called `startTime` which will be the `date` when our timer begins. The value of `startTime` will be set to the new date, giving us the current time and date. Instead of calling `setInterval` we would call `getTimerTime`, which would take a new date, which would be the current time, and subtract it from our `startTime`. This is going to be in milliseconds and we want to convert this to seconds. To do so, we will divide by 1000, which will give us the time in seconds, but it could be a decimal, so we will add `Math.floor` to ensure it is always an integer, i.e. it will always round down to 1.
```

Comment:
If possible, break down this para into smaller paras.

I believe you are explaining the code snippets, so you could use bulleted points as well